internal/controller/prefix_controller.go

```diff
@@ -81,6 +82,14 @@ func (r *PrefixReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		if controllerutil.ContainsFinalizer(prefix, PrefixFinalizerName) {
 			if !prefix.Spec.PreserveInNetbox {
 				if err := r.NetboxClient.DeletePrefix(prefix.Status.PrefixId); err != nil {
+					if errors.Is(err, utils.ErrNotFound) {
+						setConditionErr := r.SetConditionAndCreateEvent(ctx, prefix, netboxv1.ConditionPrefixReadyFalse, corev1.EventTypeWarning, err.Error())
```

Comment:
Propose to use a different condition (replace netboxv1.ConditionPrefixReadyFalse with a new condition like e.g. netboxv1.ConditionPrefixReadyFalseDeletionFailed)

Currently the condition looks as follows:

```
  conditions:
  - lastTransitionTime: "2024-10-01T07:14:46Z"
    message: 'Failed to reserve IP in NetBox. failed to fetch IPAddress: not found'
    reason: FailedToReserveIpInNetbox
    status: "False"
    type: Ready
```

Propose to use reason "FailedToDeleteIpInNetbox" and message "Failed to delete IP in NetBox".

Same applies to the other controller...

Reply:
I agree